File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];
        for (int c = 0; c < commands.length; c++) {
            int i = commands[c][0] - 1;
            int j = commands[c][1];
            int k = commands[c][2] - 1;
            
            int[] sub = Arrays.copyOfRange(array, i, j);
            Arrays.sort(sub);
            answer[c] = sub[k];
        }
        return answer;
    }
}

File: 프로그래머스/1/42748. K번째수/K번째수.js
function solution(array, commands) {
    var answer = [];
    for (let i = 0; i < commands.length; i++) {
        let command = commands[i];
        let start = command[0] - 1;
        let end = command[1];
        let k = command[2] - 1;
        
        // 배열 자르고 오름차순 정렬
        let sliced = array.slice(start, end).sort((a, b) => a - b);
        answer.push(sliced[k]);
    }
    return answer;
}
